src/Grid/topology.jl

```diff
@@ -14,65 +13,90 @@ the given entity is included in the returned list as well.
+- `cell_neighbor::AbstractArray{AbstractVector{Int}, 1}`:            cellid to all connected cells
+- `face_neighbor::AbstractArray{AbstractVector{FaceIndex}, 2}`:      `face_neighbor[cellid,   local_face_id]`   -> neighboring faces
+- `edge_neighbor::AbstractArray{AbstractVector{EdgeIndex}, 2}`:      `edge_neighbor[cellid,   local_edge_id]`   -> neighboring edges
+- `vertex_neighbor::AbstractArray{AbstractVector{VertexIndex}, 2}`:  `vertex_neighbor[cellid, local_vertex_id]` -> neighboring vertices
+- `face_skeleton::Union{Vector{FaceIndex}, Nothing}`:     List of unique faces in the grid given as `FaceIndex`
+- `edge_skeleton::Union{Vector{EdgeIndex}, Nothing}`:     List of unique edges in the grid given as `EdgeIndex`
+- `vertex_skeleton::Union{Vector{VertexIndex}, Nothing}`: List of unique vertices in the grid given as `VertexIndex`
+
+!!! warning "Limitations"
+    The implementation only works with conforming grids, i.e. grids without "hanging nodes". Non-conforming grids will give unexpected results.
+    Grids with embedded cells (different reference dimension compared
+    to the spatial dimension) are not supported, and will error on construction.
+
 """
 mutable struct ExclusiveTopology <: AbstractTopology
-    # maps a global vertex id to all cells containing the vertex
-    vertex_to_cell::Vector{Set{Int}}
-    # index of the vector = cell id ->  all other connected cells
-    cell_neighbor::Vector{EntityNeighborhood{CellIndex}}
-    # face_neighbor[cellid,local_face_id] -> exclusive connected entities (not restricted to one entity)
-    face_face_neighbor::Matrix{EntityNeighborhood{FaceIndex}}
-    # vertex_neighbor[cellid,local_vertex_id] -> exclusive connected entities to the given vertex
-    vertex_vertex_neighbor::Matrix{EntityNeighborhood{VertexIndex}}
-    # edge_neighbor[cellid,local_edge_id] -> exclusive connected entities of the given edge
-    edge_edge_neighbor::Matrix{EntityNeighborhood{EdgeIndex}}
-    # lazy constructed face topology
+    vertex_to_cell::ArrayOfVectorViews{Int, 1}
+    cell_neighbor::ArrayOfVectorViews{Int, 1}
+    # face_face_neighbor[cellid,local_face_id] -> exclusive connected entities (not restricted to one entity)
+    face_face_neighbor::ArrayOfVectorViews{FaceIndex, 2}
+    # edge_edge_neighbor[cellid,local_edge_id] -> exclusive connected entities of the given edge
+    edge_edge_neighbor::ArrayOfVectorViews{EdgeIndex, 2}
+    # vertex_vertex_neighbor[cellid,local_vertex_id] -> exclusive connected entities to the given vertex
+    vertex_vertex_neighbor::ArrayOfVectorViews{VertexIndex, 2}
+
     face_skeleton::Union{Vector{FaceIndex}, Nothing}
     edge_skeleton::Union{Vector{EdgeIndex}, Nothing}
     vertex_skeleton::Union{Vector{VertexIndex}, Nothing}
-    # TODO reintroduce the codimensional connectivity, e.g. 3D edge to 2D face
+end
+
+function ExclusiveTopology(grid::AbstractGrid{sdim}) where sdim
```

Comment:
Just want to highlight that this is more restrictive since before the argument was `Vector{C} where C <: AbstractCell`

Reply:
Good point - I've added this change (and motivation) in the changelog.